main.go

```diff
@@ -58,16 +62,76 @@ func handleJump(args []string, provider string) []string {
 	return ssh_args_str
 }
 
-func selectHost(hosts []cr.Host) cr.Host {
+func hostMatch(f *cr.Host, filterstr []string) bool {
+	val := reflect.ValueOf(f).Elem()
+	for i := 0; i < val.NumField(); i++ {
+		valueField := val.Field(i)
+		typeField := val.Type().Field(i)
+		log.Tracef("Field Name: %s,\t Field Value: %v\n", typeField.Name, valueField.Interface())
+		matched1, err1 := regexp.MatchString(filterstr[0], strings.ToLower(typeField.Name))
+		if err1 != nil {
+			log.Errorf("Error matching regexp %s: %#v", filterstr[0], err1)
+		} else {
+			if matched1 {
+				log.Tracef("Field name: %s is matching %s", typeField.Name, filterstr[0])
+				matched2, err2 := regexp.MatchString(filterstr[1], valueField.Interface().(string))
+				if err2 != nil {
+					log.Errorf("Error matching regexp %s: %#v", filterstr[1], err2)
+				} else {
+					if matched2 {
+						log.Debugf("Field value: %#v of field %s is matching %s", valueField.Interface(), typeField.Name, filterstr[1])
+						return true
+					} else {
+						log.Tracef("Field value: %#v not matching %s", valueField.Interface(), filterstr[1])
+					}
+				}
+			} else {
+				log.Tracef("Field name: %s not matching %s", typeField.Name, filterstr[0])
+			}
+		}
+	}
+	return false
+}
+
+func filterHosts(hosts []cr.Host, filter string) cr.Host {
+	host := hosts[0]
+	if filter != "" {
```

Comment:
How about doing `if filter == "" { return host}` and keep to rest of the code not a big branch ?

Reply:
Done